src/Routes.js

```diff
@@ -9,13 +9,16 @@ import { SignIn as SignInView, AccessDenied as AccessDeniedView } from './views'
 
 const Routes = (props) => {
   const { authenticated, authorized, tabs } = props;
-
+  const previousUrl = sessionStorage.getItem('previousURL');
   return (
     <Routes_>
-      {/*
-      <Route path={'/'} element={<Navigate to="/scenario" replace />}></Route>
-*/}
-      <Route path="/sign-in" element={authenticated === false ? <SignInView /> : <Navigate to="/scenario" />} />
+      <Route path={'/'} element={<Navigate to={previousUrl || '/scenario'} replace />}></Route>
+      <Route
+        path="/sign-in"
+        element={
+          authenticated === false ? <SignInView /> : <Navigate to={history.state.idx === 0 ? '/scenario' : -1} />
```

Comment:
Indeed `authenticated === false` is more explicit and readable but it is little shorter to use `!authenticated` test condition.